client/signup/main.jsx

```diff
@@ -598,9 +613,13 @@ class Signup extends Component {
 
 	getDependenciesInQuery = () => {
 		const queryObject = this.props.initialContext?.query ?? {};
-		const dependenciesInQuery =
-			flows.getFlow( this.props.flowName, this.props.isLoggedIn )?.providesDependenciesInQuery ??
-			[];
+
+		const flow = flows.getFlow( this.props.flowName, this.props.isLoggedIn );
+
+		const requiredDependenciesInQuery = flow?.providesDependenciesInQuery ?? [];
+		const optionalDependenciesInQuery = flow?.optionalDependenciesInQuery ?? [];
+
+		const dependenciesInQuery = [ ...requiredDependenciesInQuery, ...optionalDependenciesInQuery ];
```

Comment:
Not sure if we should be concerned about this, but there is existing code that uses `getDependenciesInQuery` which, according to the var name, seems to assume that it only returns dependencies defined in `providesDependenciesInQuery`:

https://github.com/Automattic/wp-calypso/blob/5c038a94ecccd9ceaab320ccf975fc9256b18e96/client/signup/main.jsx#L157

Would the addition of `optionalDependenciesInQuery` cause any unexpected side effect?

Reply:
Let's play it safe. I'll move the logic in the getRecordProps.